### PR TITLE
ROX-18106: compliance profile protos stores

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Just the protos and stores for profiles and rules.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI of generated UTs.

### CODE REVIEW

File: proto/storage/compliance_operator_v2.proto [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+// Next Tag: 12
+message ComplianceOperatorProfileV2 {
+    string              id              = 1 [(gogoproto.moretags) = 'sql:"pk"'];
+    string              profile_id      = 2;
+    string              name            = 3 [(gogoproto.moretags) = 'search:"Compliance Profile Name,hidden,store" sql:"index=category:unique;name:rule_unique_indicator"'];
+    string              version         = 4 [(gogoproto.moretags) = 'search:"Compliance Operator Version,hidden,store" sql:"index=category:unique;name:rule_unique_indicator"'];
+    string              profile_version = 5 [(gogoproto.moretags) = 'search:"Compliance Profile Version,hidden,store"'];
+    string              product_type    = 6 [(gogoproto.moretags) = 'search:"Compliance Operator Profile Product,hidden,store"'];
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:18]: 1:1 mapping will not hold true in the case of composite profiles. I recommend an array.

File: proto/storage/compliance_operator_v2.proto [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    string              standard        = 7 [(gogoproto.moretags) = 'search:"Compliance Operator Profile Standard,hidden,store"'];
+    map<string, string> labels          = 8;
+    map<string, string> annotations     = 9;
+    string              description     = 10;
+    message Rule {
+        string rule_name       = 1 [(gogoproto.moretags) = 'search:"-" sql:"fk(ComplianceOperatorRuleV2:name),no-fk-constraint"'];
+    }
+    repeated Rule       rules       = 11;
+}
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:27]: Let's capture the product as column (?)

File: proto/storage/compliance_operator_v2.proto [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+// Next Tag: 13
+message ComplianceOperatorRuleV2 {
+    string              name         = 1 [(gogoproto.moretags) = 'search:"Compliance Rule Name,hidden,store" sql:"pk"'];
+    string              version      = 2 [(gogoproto.moretags) = 'search:"Compliance Operator Version,hidden,store"'];
+    string              rule_version = 3 [(gogoproto.moretags) = 'search:"Compliance Rule Version,hidden,store"'];
+    string              product_type = 4 [(gogoproto.moretags) = 'search:"Compliance Operator Rule Product,hidden,store"'];
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:34]: Are you planning on deriving `product_type` from profile?

File: proto/storage/compliance_operator_v2.proto [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    string              rule_type    = 5 [(gogoproto.moretags) = 'search:"Compliance Operator Rule Type,hidden,store"'];
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:35]: What is a rule type?

**Review by md2119 — COMMENTED** [review]
I made some recommendations and left some clarifying questions.

File: proto/storage/compliance_operator_v2.proto (hunk 3 above)
Comment (dashrews78) [reply proto/storage/compliance_operator_v2.proto:34]: I just took a closer look at the comments on the data model.  I'm going to take this out.  
When we get to doing tailored profiles we will have to re-evaluate if we need something there or not.

File: proto/storage/compliance_operator_v2.proto (hunk 4 above)
Comment (dashrews78) [reply proto/storage/compliance_operator_v2.proto:35]: Node, Platform, Workload, etc.  It is the `CheckType` field in the CR I'm looking at.

File: proto/storage/compliance_operator_v2.proto (hunk 2 above)
Comment (dashrews78) [reply proto/storage/compliance_operator_v2.proto:27]: I'm not tracking this question.  What do you mean by "product"?

Comment (md2119) [reply proto/storage/compliance_operator_v2.proto:27]: `compliance.openshift.io/product: redhat_openshift_container_platform_4.1`

Comment (dashrews78) [reply proto/storage/compliance_operator_v2.proto:27]: got it.  sure.

**Review by md2119 — APPROVED** [review]

**Pull request merged**